kedro-datasets/tests/conftest.py

```diff
@@ -32,3 +41,87 @@ def save_args(request):
 @fixture(params=[None])
 def fs_args(request):
     return request.param
+
+
+@fixture(params=[None])
+def mock_fs_args(request):
+    fs_args = {
+        # NB: use moto server to mock S3
+        "client_kwargs": {"endpoint_url": ENDPOINT_URI}
+    }
+
+    if isinstance(request.param, dict):
+        fs_args.update(request.param)
+
+    return fs_args
+
+
+@fixture
+def credentials():
+    return {
+        "key": "fake_access_key",
+        "secret": "fake_secret_key",
+    }
```

Comment:
Why there are 2 sets of credentials?

Do we need `key` and `secret` or the `AWS_XXXXX`?

Reply:
One is for credentials file, the other one are env vars. The env vars are required for bucket setup etc while the credentials are used in the Dataset classes.